moqt-server/src/modules/object_cache_storage.rs

```diff
@@ -0,0 +1,2091 @@
+            .set_channel(session_id, subscribe_id, header.clone())
+            .await;
+
+        for j in 0..10 {
+            let group_size = 15;
+
+            for i in 0..group_size {
+                let object_payload: Vec<u8> = vec![
+                    j * group_size + i,
+                    j * group_size + i + 1,
+                    j * group_size + i + 2,
+                    j * group_size + i + 3,
+                ];
+                let object_id = i as u64;
+
+                let subgroup =
+                    ObjectStreamSubgroup::new(object_id, object_status, object_payload).unwrap();
+
+                let cache_object = CacheObject::Subgroup(subgroup.clone());
+
+                let _ = object_cache_storage
+                    .set_object(session_id, subscribe_id, cache_object, duration)
+                    .await;
+            }
+        }
+
+        let expected_object_id = 0;
+        let expected_object_payload = vec![0, 1, 2, 3];
+        let expected_subgroup =
+            ObjectStreamSubgroup::new(expected_object_id, object_status, expected_object_payload)
+                .unwrap();
+
+        let result = object_cache_storage
+            .get_latest_group(session_id, subscribe_id)
+            .await;
+
+        assert!(result.is_ok());
+
+        let result_object = match result.unwrap().unwrap() {
+            (_, CacheObject::Subgroup(object)) => object,
+            _ => panic!("cache object not matched"),
+        };
+
+        assert_eq!(result_object, expected_subgroup);
+    }
+
+    #[tokio::test]
+    async fn delete_channel() {
+        let session_id = 0;
+        let subscribe_id = 1;
+        let track_alias = 3;
+        let group_id = 4;
+        let subgroup_id = 5;
+        let publisher_priority = 6;
+        let header = CacheHeader::Subgroup(
+            StreamHeaderSubgroup::new(
+                subscribe_id,
+                track_alias,
+                group_id,
+                subgroup_id,
+                publisher_priority,
+            )
+            .unwrap(),
+        );
+
+        // start object cache storage thread
+        let (cache_tx, mut cache_rx) = mpsc::channel::<ObjectCacheStorageCommand>(1024);
+        tokio::spawn(async move { object_cache_storage(&mut cache_rx).await });
+
+        let mut object_cache_storage = ObjectCacheStorageWrapper::new(cache_tx);
+
+        let _ = object_cache_storage
+            .set_channel(session_id, subscribe_id, header.clone())
+            .await;
+
+        let delete_result = object_cache_storage
+            .delete_channel(session_id, subscribe_id)
+            .await;
+
+        assert!(delete_result.is_ok());
+
+        let get_result = object_cache_storage
+            .get_header(session_id, subscribe_id)
+            .await;
+
+        assert!(get_result.is_err());
+    }
+
+    #[tokio::test]
+    async fn get_latest_group_ascending_track_long() {
+        let object_payload = vec![
```

Comment:
Why can't we use shorter object_payload?

If it is unnecessary, we can make code more simple

Reply:
I will delete this test.